code/modules/mob/living/carbon/xenomorph/life.dm

```diff
@@ -139,24 +139,20 @@
 	var/turf/T = loc
 	if(!T || !istype(T))
 		return
-	if(current_aura)
-		plasma_stored -= 5
-	if(plasma_stored == xeno_caste.plasma_max)
-		return
 	if(locate(/obj/effect/alien/weeds) in T)
 		plasma_stored += xeno_caste.plasma_gain
 		if(recovery_aura)
 			plasma_stored += round(xeno_caste.plasma_gain * recovery_aura * 0.25) //Divided by four because it gets massive fast. 1 is equivalent to weed regen! Only the strongest pheromones should bypass weeds
 	else
 		plasma_stored++
+
+	if(current_aura && (plasma_stored -= 5) < 5)
+		plasma_stored = 0
+        current_aura = null
+        to_chat(src, "<span class='warning'>You have run out of plasma and stopped emitting pheromones.</span>")
+
 	if(plasma_stored > xeno_caste.plasma_max)
 		plasma_stored = xeno_caste.plasma_max
```

Comment:
Needs a return after this. We dont need to go to hud_set_plasma if it is already at or above max.
It is unlikely apart from a edgecase of specific numbers that its needed to update.